components/common/src/dynamo/common/config_dump/config_dumper.py

```
@@ -0,0 +1,186 @@
+            dump_path.parent.mkdir(parents=True, exist_ok=True)
+            with open(dump_path.resolve(), "w+") as f:
+                f.write(debug_config_payload)
+            logger.info(f"Dumped debug config to {dump_path.resolve()}")
+        except Exception as e:
+            logger.error(f"Failed to dump debug config to {dump_config_to}: {e}")
+            logger.info("DEBUG_CONFIG:" + debug_config_payload)
+    else:
+        logger.info("DEBUG_CONFIG:" + debug_config_payload)
+
+
+def get_config_dump(config: Any, extra_info: Dict[str, str] = {}) -> str:
```

Ruff (0.13.1)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)